File: 278-first-bad-version/278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        low: int = 1
        high: int = n
        
        while(low <= high):
            middle: int = low + int((high - low)/2);
            if isBadVersion(middle):
                correct_version: int = middle
                high = middle - 1
            else:
                low = middle + 1
            
        return correct_version

File: 278-first-bad-version/278-first-bad-version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int low = 1;
        int high = n;
        int middle;
        int correctVersion;
        while(low <= high){
            middle = low + (high - low)/2;
            if(!isBadVersion(middle)){
                low = middle + 1;
            }
            else{
                correctVersion = middle;
                high = middle - 1;  
            }
        }
        return correctVersion;
        
    }
};